### PR TITLE
Realize participants leaving

### PR DESCRIPTION
**Pull request by HaiyueZhang**
Add participants leaving UI
Realize removing participants from database.

### CODE REVIEW

File: src/pages/event/[id].tsx [hunk 1]
```diff
@@ -41,16 +45,18 @@ const EventInfoHeader: React.FC = () => {
 
   const [isDeleteDialogOpen, setIsDeleteDialogOpen] = useState(false);
   const [isEditDialogOpen, setIsEditDialogOpen] = useState(false);
+  const [isLeaveDialogOpen, setIsLeaveDialogOpen] = useState(false);
 
   const editEventName = api.events.updateEvent_name.useMutation();
   const deleteEvent = api.events.deleteEvent.useMutation();
+  const deleteParticipate = api.participates.deleteParticipate.useMutation();
   const editEventParticipants =
```
Comment (louisunlimited) [inline src/pages/event/[id].tsx:53]: Does `deleteParticipate` remove all the time slots as well?

**Review by louisunlimited — CHANGES_REQUESTED** [review]

File: src/pages/event/[id].tsx (hunk 1 above)
Comment (HaiyueZhang) [reply src/pages/event/[id].tsx:53]: I do not think so.

File: src/components/section/Dialog.tsx [hunk 2]
```diff
@@ -259,3 +259,98 @@ export const DeleteDialog: React.FC<DeleteDialogProps> = ({
...
+              <Dialog.Title className="text-left text-lg font-semibold leading-6 text-gray-900">
+                Confirm Leave Event
+              </Dialog.Title>
+              <div className="mt-4 text-left text-2xl font-bold">
+                {eventName}
+              </div>
+              <div className="mt-2 text-left text-xs text-gray-500">
+                You will leave the event permanently, including all availability
+                data. This action is irreversible and can not be undone.
+              </div>
```
Comment (louisunlimited) [inline src/components/section/Dialog.tsx:319]: We should probably inform the user that the time slots will be deleted as well, and prob make this action more informative with event name included. prob sth like `You will leave <event name> permanently, your selected time slots will also be deleted. This action is irreversible and can not be undone.`

We could also add a typing check like when you delete projects on Vercel, but I guess we don't have to do that for participants leaving events but we can consider doing this for when owner deletes the whole event

**Review by louisunlimited — CHANGES_REQUESTED** [review]

**Review by RaymondWHZ — APPROVED** [review]
Really nice work!

**Pull request merged**